handler/category-dimension-import.go

```diff
@@ -2,32 +2,73 @@ package handler
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
+	"github.com/ONSdigital/dp-api-clients-go/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/dataset"
 	"github.com/ONSdigital/dp-import-cantabular-dimension-options/config"
 	"github.com/ONSdigital/dp-import-cantabular-dimension-options/event"
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
 // CategoryDimensionImport is the handle for the CategoryDimensionImport event
 type CategoryDimensionImport struct {
-	cfg config.Config
+	cfg      config.Config
+	ctblr    CantabularClient
+	datasets DatasetAPIClient
 }
 
-// NewCategoryDimensionImport creates a new CategoryDimensionImport with the provided config
-func NewCategoryDimensionImport(cfg config.Config) *CategoryDimensionImport {
+// NewCategoryDimensionImport creates a new CategoryDimensionImport with the provided config and Cantabular client
+func NewCategoryDimensionImport(cfg config.Config, c CantabularClient, d DatasetAPIClient) *CategoryDimensionImport {
 	return &CategoryDimensionImport{
-		cfg: cfg,
+		cfg:      cfg,
+		ctblr:    c,
+		datasets: d,
 	}
 }
 
-// Handle takes a single event
+// Handle calls Cantabular server to obtain a list of variables for a CantabularBlob,
+// which are then posted to the dataset API as dimension options
 func (h *CategoryDimensionImport) Handle(ctx context.Context, e *event.CategoryDimensionImport) error {
 	logData := log.Data{
 		"event": e,
 	}
 	log.Info(ctx, "event handler called", logData)
 
-	// TODO process message
+	// obtain the possible values for the provided dimension and blob
+	resp, err := h.ctblr.GetCodebook(ctx, cantabular.GetCodebookRequest{
+		DatasetName: e.CantabularBlob,
+		Categories:  true,
+		Variables:   []string{e.DimensionID},
+	})
+	if err != nil {
+		return err
```

Comment:
I think it's always better to wrap errors with context at each level rather than return raw